### PR TITLE
Add 'unix' condition

### LINKED ISSUE #166: testscript: add 'unix' condition?
[Now](https://go-review.googlesource.com/c/go/+/389934/) that Go considers the `unix` pseudo-tag to be satisfied by any sufficiently Unixy system, following a successful [campaign](https://github.com/golang/go/issues/20322) by myself (and, to a lesser extent, Rob Pike)—should `testscript`'s imported copy of `build.go` be updated to support this as a condition in scripts?

Then I could write:

```
[unix] exec echo 'It''s a UNIX system! I know this!'
```

### PR DESCRIPTION
**Pull request by bitfield**
(see commit messages)

Fixes #166.

### CODE REVIEW

**Review by mvdan — COMMENTED** [review]
Change itself SGTM - I didn't notice that the goos and goarch conditions weren't documented yet.

As for the test, I would say use a "nested" testscript test. See the testdata changes in https://github.com/rogpeppe/go-internal/pull/167/files, for example. You could check that `[unix]` matches when the host is `[linux]` and not when it's `[windows]`, for example - that should keep things simple enough and get us covered with CI.

**Comment by bitfield** [comment]
Sorry, I'm not completely clear on what you're suggesting. Could you elaborate a little?

> You could check that [unix] matches when the host is [linux] and not when it's [windows]

That was the aim of the two tests guarded by mutually exclusive build tags—I couldn't find a way to get `testscript` to "pretend" to be running on a particular GOOS. Setting `env GOOS=linux` in the script has no effect. Is there a neater way to do this?

**Comment by mvdan** [comment]
Just pushed a commit to show what I mean :) I ended up deciding against using nested testscript, because we really don't need it. The end result is very close to what you had done, but with far less test code, thanks to testscript itself. The test only checks linux/darwin/windows, but I reckon that is enough given how popular they are, and that they are the three OSes that we use CI on.

**Comment by mvdan** [comment]
Oh, and while there, I also rewrote the existing cond.txt to use the native `mkdir` and `exists` rather than an external shell, which means more portability - particularly necessary for Windows. So perhaps we should wait for an extra review from @myitcv or @rogpeppe since I can't really approve my own changes there.

**Comment by bitfield** [comment]
Great! Very good suggestions, thank you @mvdan.

**Comment by mvdan** [comment]
If it sounds good to you, then squash into your second commit - then we're ready for another review and a merge.

**Review by myitcv — APPROVED** [review]
LGTM with one question.

File: imports/build.go [hunk 1]
```diff
@@ -195,17 +195,26 @@ func MatchFile(name string, tags map[string]bool) bool {
...
+		UnixOS[v] = true
+	}
 	for _, v := range strings.Fields(goarchList) {
 		KnownArch[v] = true
 	}
 }
 
-const goosList = "aix android darwin dragonfly freebsd hurd illumos ios js linux nacl netbsd openbsd plan9 solaris windows zos "
-const goarchList = "386 amd64 amd64p32 arm armbe arm64 arm64be loong64 mips mipsle mips64 mips64le mips64p32 mips64p32le ppc ppc64 ppc64le riscv riscv64 s390 s390x sparc sparc64 wasm "
+const (
```
[the remark this answers is not in the document]
Comment (bitfield) [reply imports/build.go:216]: Great question! The comment at the top of this file says:

```go
// Copied from Go distribution src/go/build/build.go, syslist.go
```

But that's a long way from this information. Do you think I should add a comment attached to these constants specifically to say that they come from `syslist.go`, and that they should track updates to that file?

[the remark this answers is not in the document]
Comment (mvdan) [reply imports/build.go:218]: I'm satisfied with the added comment, resolving and merging :)

**Review by mvdan — APPROVED** [review]

**Pull request merged**